src/translate.service.ts

```diff
@@ -286,9 +286,12 @@ export class TranslateService {
      * @param interpolateParams
      * @returns {any}
      */
-    public getParsedResult(translations: any, key: any, interpolateParams?: Object): any {
+    public getParsedResult(translations: any, key: any, interpolateParams?: any): any {
         let res: string|Observable<string>;
 
+        interpolateParams = interpolateParams || {};
+        interpolateParams.currentLang = interpolateParams.currentLang || this.currentLang;
```

Comment:
now it should be ok, see the use case https://github.com/aitboudad/ng2-translate/blob/dev/src/translate.parser.ts#L58-L79